src/main/java/nl/tudelft/ti2206/group9/level/Track.java

```diff
@@ -78,14 +78,32 @@ public final synchronized void moveTrack(final double dist) {
 							< GameScreen.CAMERA_TRANS.getZ()) {
 						entity.selfDestruct();
 					}
-					entity.getCenter().addZ(-dist);
-					entity.checkCollision(entities.get(player));
+					moveEntity(entity, -dist);
 				}
 			}
 		}
 	}
 
 	/**
+	 * Make sure the collisions are checked over the interval that the entities
+	 * are moved.
+	 * @param first Entity that the collision is checked for.
```

Comment:
This parameter has been removed from the method. Please refactor.

< documentation